File: docs/components_page/components/dropdown/menu_items.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

dropdown <- htmlDiv(
  list(
    dbcDropdownMenu(
      list(
        dbcDropdownMenuItem("A button", id = "dropdown-button", n_clicks = 0),
        dbcDropdownMenuItem(
          "Internal link",
          href = "/docs/components/dropdown_menu"
        ),
        dbcDropdownMenuItem(
          "External Link",
          href = "https://github.com"
        ),
        dbcDropdownMenuItem(
          "External relative",
          href = "/docs/components/dropdown_menu",
          external_link = TRUE
        )
      ),
      label = "Menu"
    ),
    htmlP(id = "item-clicks", className = "mt-3")
  )
)


app$callback(
  output("item-clicks", "children"),
  list(input("dropdown-button", "n_clicks")),
  function(n) {
    if (n > 0) {
      return(sprintf("Button clicked %d times.", n))
    }
    return(sprintf("Button not clicked yet"))
  }
)

File: docs/components_page/components/dropdown/menu_items.jl
using DashBootstrapComponents, DashHtmlComponents

dropdown = html_div([
    dbc_dropdownmenu(
        [
            dbc_dropdownmenuitem("A button", id = "dropdown-button", n_clicks = 0),
            dbc_dropdownmenuitem("Internal link", href = "/docs/components/dropdown_menu"),
            dbc_dropdownmenuitem("External Link", href = "https://github.com"),
            dbc_dropdownmenuitem(
                "External relative",
                href = "/docs/components/dropdown_menu",
                external_link = true,
            ),
        ],
        label = "Menu",
    ),
    html_p(id = "item-clicks", className = "mt-3"),
]);

callback!(app, Output("item-clicks", "children"), Input("dropdown-button", "n_clicks")) do n
    return n > 0 ? "Button clicked $n times." : "Button not clicked yet."
end;
